toolchain/semantics/semantics_node_block_stack.h

```diff
@@ -23,9 +23,18 @@ class SemanticsNodeBlockStack {
       llvm::raw_ostream* vlog_stream)
       : name_(name), node_blocks_(&node_blocks), vlog_stream_(vlog_stream) {}
 
+  // Adds a node block, but does not push it.
+  auto Add() -> SemanticsNodeBlockId;
```

Comment:
My leaning is this belongs on SemanticsIR which owns node_blocks_, rather than here, since it's not related to the stack. This could switch to taking a pointer to semantics_ir for calling Add.

SemanticsIR already has Add functions for most of its data, so this would be consistent there. To the extent that the stack was abstracting away the Add, that's no longer possible with this addition.

Reply:
Done.